web/api/v1/api.go

```diff
@@ -538,14 +546,54 @@ func (api *API) labelValues(r *http.Request) (result apiFuncResult) {
 		q.Close()
 	}
 
-	vals, warnings, err := q.LabelValues(name)
+	var vals []string
+	var warnings storage.Warnings
+	if len(r.Form["match[]"]) > 0 {
+		// Get all series which match matchers.
+		var sets []storage.SeriesSet
+		for _, mset := range matcherSets {
+			s := q.Select(false, nil, mset...)
+			sets = append(sets, s)
+		}
+		vals, warnings, err = labelValuesByMatchers(sets, name)
+	} else {
+		vals, warnings, err = q.LabelValues(name)
+	}
+
 	if err != nil {
 		return apiFuncResult{nil, &apiError{errorExec, err}, warnings, closer}
 	}
 
 	return apiFuncResult{vals, nil, warnings, closer}
 }
 
+// LabelValuesByMatchers uses matchers to filter out matching series, then label values are extracted.
+func labelValuesByMatchers(sets []storage.SeriesSet, name string) ([]string, storage.Warnings, error) {
+	set := storage.NewMergeSeriesSet(sets, storage.ChainedSeriesMerge)
+	labelValuesMap := make(map[string]string)
+	for set.Next() {
+		series := set.At()
+		labelValue := series.Labels().Get(name)
+		// This is to avoid duplicates.
```

Comment:
You can do a simple assignment and not worry about it.

Reply:
True, fixed